core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -143,6 +143,21 @@ trait WorkspaceSupport {
     }
   }
 
+  def withWorkspaceBucketRegionCheck[T](bucketRegion: Option[String])(op: => Future[T]): Future[T] = {
```

Comment:
I'm curious why we wrote this as a 'with' as opposed to just a plain function that validates and possibly throws? We don't seem to be doing anything that requires Futures here.

Reply:
It's following what it's [surrounding code is doing](https://github.com/broadinstitute/rawls/blob/360fab8c12f920b6102415947f9f588b86d5cc60/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L176-L183) as well as the naming convention in Rawls when you want your code to do stuff and then run the block of code passed to it as argument. So here we say check the bucket region and then run the code block specified as `op: => Future[T]`. So I personally understand it as run the code with bucket region check.